### PR TITLE
Broadcasting

### PR DESCRIPTION
**Pull request by Rhuansantos**
This Pull Request is implementing the !broadcast feature that allows the user send messages for all channels or for the select ones!
Also, it gives a little Improve to the !help message string

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,5 +27,15 @@
     "eslint-plugin-react": "^7.1.0",
     "mocha": "^3.5.0",
     "mocha-eslint": "^4.1.0"
+  },
+  "peerDependencies": {
```
Comment (reactivepixel) [inline package.json]: Let's remove these from the pr

**Comment by reactivepixel** [comment]
Commenting on Broadcast.js controller is a bit sparse.

**Comment by reactivepixel** [comment]
!help changes should be documented.

File: bot/controllers/broadcast.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  const messageMiddleware = (message) => {
+    const container = {};
+    container.parsed = message.content.split(' ');
+    const msg = Object.assign(message, container);
+    return msg;
+  };
+  // send message for all channels
+  const _allChannels = (message) => {
+    const msg = messageMiddleware(message);
+    const input = msg.parsed.join(',').replace(/,/g, ' ');
```
Comment (reactivepixel) [inline bot/controllers/broadcast.js]: Semantically, input is a bit too generically named.

File: bot/controllers/broadcast.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    if (msg.parsed[0].toLowerCase() === '!broadcastall') {
+      const channels = message.guild.channels.map(channel => channel);
+      Object.keys(channels).forEach((el) => {
+        // filter for text channels
+        if (channels[el].type === 'text') {
+          // creating an object to send a message
+          const objEmbed = new Discord.RichEmbed()
```
Comment (reactivepixel) [inline bot/controllers/broadcast.js]: Semantically objEmbed is, again, generic. Try to find a name that fits the context of this situation of its use.

File: bot/controllers/broadcast.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+      // Verify if the user choose a valid channel if yes display the message
+      const isValid = inputChannels.split(',').every(r => validChannels.includes(r));
+      if (isValid) {
+        inputChannels.split(',').forEach((el) => {
+          // message.guild.channels.find('name', el).send(input);
+          const objEmbed = new Discord.RichEmbed()
+            .setAuthor(message.author.tag, message.author.displayAvatarURL)
+            // remove the first index of the array in that case are the channels
+            .setTitle(`:loudspeaker: ${input}`)
+            .setColor([145, 124, 57]);
```
Comment (reactivepixel) [inline bot/controllers/broadcast.js]: This is the second time using this exact color. Would it not be better represented as a variable defined toward the top of the page/function?

Comment (Rhuansantos) [reply bot/controllers/broadcast.js]: Agree on that @reactivepixel!

File: bot/controllers/broadcast.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+          message.guild.channels.find('name', el).send(objEmbed);
+        });
+      } else {
+        message.reply('you choose invalid channels');
```
Comment (reactivepixel) [inline bot/controllers/broadcast.js]: Choose is the wrong tense. It's small, but items like are the interface that the user actually sees so these end up mattering more than they should.

Chose is the past tense version that should be implemented.

File: bot/controllers/broadcast.js (hunk 2 above)
Comment (ghost) [reply bot/controllers/broadcast.js]: Would suggest a name like `userMessageInput`.

Comment (ghost) [reply bot/controllers/broadcast.js]: or `usrMsgInput`.

File: bot/controllers/broadcast.js (hunk 3 above)
Comment (ghost) [reply bot/controllers/broadcast.js]: would suggest something like `botMsg` or something that lets us know what it is and who/what is sending it.

File: bot/controllers/broadcast.js (hunk 2 above)
Comment (Rhuansantos) [reply bot/controllers/broadcast.js]: Thanks for the suggestion @joegreen2!

File: bot/controllers/broadcast.js (hunk 5 above)
Comment (ghost) [reply bot/controllers/broadcast.js]:
```
// Small spelling error
message.reply('you choose invalid channels');


// Correction
message.reply('you chose invalid channels');
```

**Comment by Rhuansantos** [comment]
@reactivepixel Changes were made based on suggestions, anything missed?

**Comment by reactivepixel** [comment]
👍 LGTM 🍻

**Pull request closed without merging**